aten/src/ATen/templates/TensorBody.h

```diff
@@ -242,22 +248,67 @@ class TORCH_API Tensor: public TensorBase {
   C10_DEPRECATED_MESSAGE("packed_accessor is deprecated, use packed_accessor32 or packed_accessor64 instead")
   GenericPackedTensorAccessor<T,N,PtrTraits,index_t> packed_accessor() && = delete;
 
-  Tensor operator~() const;
-  Tensor operator-() const;
-  Tensor& operator+=(const Tensor & other);
-  Tensor& operator+=(Scalar other);
-  Tensor& operator-=(const Tensor & other);
-  Tensor& operator-=(Scalar other);
-  Tensor& operator*=(const Tensor & other);
-  Tensor& operator*=(Scalar other);
-  Tensor& operator/=(const Tensor & other);
-  Tensor& operator/=(Scalar other);
-  Tensor& operator&=(const Tensor & other);
-  Tensor& operator|=(const Tensor & other);
-  Tensor& operator^=(const Tensor & other);
-  Tensor operator[](Scalar index) const;
-  Tensor operator[](Tensor index) const;
-  Tensor operator[](int64_t index) const;
+  Tensor operator~() const {
+    return bitwise_not();
+  }
+  Tensor operator-() const {
+    return neg();
+  }
+  Tensor& operator+=(const Tensor & other) {
+    return add_(other);
+  }
+  Tensor& operator+=(Scalar other) {
+    return add_(other);
+  }
+  Tensor& operator-=(const Tensor & other) {
+    return sub_(other);
+  }
+  Tensor& operator-=(Scalar other) {
+    return sub_(other);
+  }
+  Tensor& operator*=(const Tensor & other) {
+    return mul_(other);
+  }
+  Tensor& operator*=(Scalar other) {
+    return mul_(other);
+  }
+  Tensor& operator/=(const Tensor & other) {
+    return div_(other);
+  }
+  Tensor& operator/=(Scalar other) {
+    return div_(other);
+  }
+  Tensor& operator&=(const Tensor & other) {
+    return bitwise_and_(other);
+  }
+  Tensor& operator|=(const Tensor & other) {
+    return bitwise_or_(other);
+  }
+  Tensor& operator^=(const Tensor & other) {
+    return bitwise_xor_(other);
+  }
+  Tensor operator[](Scalar index) const {
+    if (!index.isIntegral(false)) {
+      TORCH_CHECK_INDEX(false, "Can only index tensors with integral scalars");
+    }
+    return select(0, index.toLong());
```

Comment:
nit: I know it was here before, but to be consistent with the one below, this could do `return this->operator[](index.toLong())`;